rust/datafusion/src/datasource/datasource.rs

```diff
@@ -20,13 +20,13 @@
 use std::sync::{Arc, Mutex};
 
 use arrow::datatypes::Schema;
+use arrow::record_batch::SendableRecordBatchReader;
 
 use crate::error::Result;
-use crate::execution::physical_plan::BatchIterator;
 
-/// Returned by implementors of `Table#scan`, this `BatchIterator` is wrapped with
+/// Returned by implementors of `Table#scan`, this `SendableRecordBatchReader` is wrapped with
 /// an `Arc` and `Mutex` so that it can be shared across threads as it is used.
-pub type ScanResult = Arc<Mutex<dyn BatchIterator>>;
+pub type ScanResult = Arc<Mutex<dyn SendableRecordBatchReader>>;
```

Comment:
Do we still need to wrap it with Arc and Mutex when `SendableRecordBatchReader` already implements Send and Sync? It looks like `ScanResult` can just be an alias to `SendableRecordBatchReader` or `Arc<Mutex<dyn RecordBatchReader>>`?

Reply:
Yea I was wondering the same thing, but it's a little out of scope for this PR.  At least the new name kinda highlights this, we can address as a follow up PR.